packages/ChangesReloaded-Core.package/CRBrowser.class/instance/getHistoryListFrom..st

```diff
@@ -0,0 +1,22 @@
+initializing
+getHistoryListFrom: anOrderedCollectionOfChangeRecords
+
+	| historyList |
+	historyList := OrderedCollection new.
+	anOrderedCollectionOfChangeRecords
+		do: [:each | each isMethod
+				ifTrue: [
+					| methodSelector methodClass historyListItemCandidates historyListItem changeListItem |
+					methodSelector := each methodSelector.
+					methodClass := each methodClassName.
+					historyListItemCandidates := historyList select: [:changeInHistory |
+						changeInHistory selector == methodSelector].
+					changeListItem := CRChangeListItem new changeRecord: each.
+					historyListItemCandidates
+						ifEmpty: [
+							historyListItem := CRHistoryListItem newWithSelector: methodSelector class: methodClass andChange: changeListItem.
+							historyList add: historyListItem];
+						ifNotEmpty: [
+							historyListItem := historyListItemCandidates first.
+							historyListItem history add: changeListItem]]].
+	^ historyList
```

Comment:
Phew, took me some time to figure out what this method does. Maybe we should first save the history items in a map/dictionary and convert it to a list at the end? That would make our code more performant because we could get the history item for a given selector in O(1), but more importantly, it would give us nice helper methods like `at: ifAbsent:`. This is some pseudo-smalltalk-y code of how that might look:
```smalltalk
historyDict := Dictionary new     (or however this looks in Smalltalk)
anOrderedCollectionOfChangeRecords
  select: [:each | each isMethod]
  do: [:each |
    | historyItem |
    historyItem := historyDict at: each methodSelector
      ifAbsent: [CRHistoryListItem newWithSelector: each methodSelector andClass: each methodClass].
    historyItem history add: changeListItem]
^ historyDict asList    (not sure if this is sorted by insertion order in Smalltalk)
```